ml4bc.ipynb

```diff
@@ -0,0 +1,392 @@
+    "            nn.Conv3d(1, 8, kernel_size=(3, 3, 3), padding=(1, 1, 1)),\n",
+    "            nn.ReLU(True),\n",
+    "            nn.Conv3d(8, 16, kernel_size=(3, 3, 3), padding=(1, 1, 1)),\n",
+    "            nn.ReLU(True),\n",
+    "        )\n",
+    "\n",
+    "        # Bottleneck (no further reduction of dimensions)\n",
+    "\n",
+    "        # Decoder\n",
+    "        self.decoder = nn.Sequential(\n",
+    "            nn.ConvTranspose3d(16, 8, kernel_size=(3, 3, 3), padding=(1, 1, 1)),\n",
+    "            nn.ReLU(True),\n",
+    "            nn.ConvTranspose3d(8, 1, kernel_size=(3, 3, 3), padding=(1, 1, 1)),\n",
+    "            nn.Sigmoid()\n",
+    "        )\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        encoded = self.encoder(x)\n",
+    "        decoded = self.decoder(encoded)\n",
+    "        return decoded\n",
+    "\n",
+    "# Create autoencoder model and use all available GPUs\n",
+    "if torch.cuda.device_count() > 1:\n",
+    "    print(\"Using\", torch.cuda.device_count(), \"GPUs!\")\n",
+    "    autoencoder = nn.DataParallel(CompactAutoencoder()).to(device)\n",
+    "else:\n",
+    "    autoencoder = CompactAutoencoder().to(device)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "513fcced-f2d8-46ce-96e3-3a667aff169a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class NetCDFDataset(Dataset):\n",
+    "    def __init__(self, root_dir, start_date, end_date, transform=True):\n",
+    "        self.root_dir = root_dir\n",
+    "        self.file_list = self.create_file_list(root_dir, start_date, end_date)\n",
+    "        self.mean = 278.83097 #279.9124\n",
+    "        self.std = 56.02780 #107.1107\n",
+    "        self.transform = transform\n",
+    "\n",
+    "    @staticmethod\n",
+    "    def create_file_list(root_dir, start_date, end_date):\n",
+    "        file_list = []\n",
+    "        time_step = timedelta(days=1)\n",
+    "        current_date = start_date\n",
+    "\n",
+    "        while current_date <= end_date:\n",
+    "            for hh in ['00', '06', '12', '18']:\n",
+    "                filename = f'{os.path.basename(root_dir)}.t2m.{current_date.strftime(\"%Y%m%d\")}{hh}.nc'\n",
+    "                file_list.append(filename)\n",
+    "            current_date += time_step\n",
+    "        \n",
+    "        return file_list\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return len(self.file_list)\n",
+    "\n",
+    "    def __getitem__(self, idx):\n",
+    "        file_path = os.path.join(self.root_dir, self.file_list[idx])\n",
+    "        \n",
+    "        # Load NetCDF data\n",
+    "        dataset = nc.Dataset(file_path)\n",
+    "        data = dataset.variables['t2m'][:].astype(np.float32)  # Adjust 'data' to the variable name in your file\n",
+    "        dataset.close()\n",
+    "        \n",
+    "        # Reshape the data to (1, 50, 721, 1440)\n",
```

Comment:
Can we make "50" as an argument that can be set up at run time?

Reply:
Its not like other models, first we need to make the dataset in that format. Then we can increase the channel. Once a model is trained you cannot do any changes to the structure, otherwise you need to train it again. I expect my training process would be done by this weekend. Tomorrow, we can talk about our future directions and updates.

Reply:
Ok, thanks